mllib/src/main/scala/org/apache/spark/ml/Predictor.scala

```diff
@@ -121,9 +121,9 @@ abstract class Predictor[
    * and put it in an RDD with strong types.
    */
   protected def extractLabeledPoints(dataset: Dataset[_]): RDD[LabeledPoint] = {
-    dataset.select(col($(labelCol)).cast(DoubleType), col($(featuresCol))).rdd.map {
```

Comment:
You need the `col($(labelCol)).cast(DoubleType)` since now all `Predictor`s take any `NumericType` for labels, see #10355